### DISCUSSION TITLE
[Feature] Allow all members of a shared album to remove at least their own added assets, if they are not the album owner, without having to delete the assets

### DISCUSSION THREAD

**Discussion by jon-f-novabackup** [immich-app/immich#2305] [Feature Request]
Currently in 1.54.0 of the server and web version (running on Ubuntu 22.10 x64 an RPi4 with ml disabled), and in the 1.54.0 build.77 of the mobile (Android) app, only the shared album owner can remove photos from a shared album. So if UserA (admin) is the owner of a shared album and UserB is just a member then if UserB logs in and uploads or adds photos to the shared album, that UserB can't then simply remove photos from the album without using the delete feature (using the remove photos from the album feature in the toolbar of the mobile app) as the icon won't display in any orientation, even when only selecting one photo that they just successfully added to the shared album). If he accidently adds items to a shared album the only recourse he has is to use the perm delete. If UserA (the admin user) tries to utilize the remove from album feature in a shared album that he does not own the same is true. Neither the website remove from album or the mobile app work in this case, unless you are the album owner. I tested this capability in google shared albums and it works in this regards to allow any member of the shared album to be able to remove a photo, even if they are not the user that uploaded that item, and even if they are not the shared album owner. I would think Immich could be made to at least allow A. the user that added the asset to the shared album to do a remove asset from album, without deleting the asset at the same time, and B. the single admin user should always be able to remove an asset from a shared album regardless of what user added it.

Back on the Immich topic, UserB's only recourse then, like if he accidently added a photo that he didn't want to, since he doesn't own the shared album, is to utilize the delete item permanently feature but the problem with that feature is currently that removes the asset from the shared album and the local device (timeline) at the same time, which is not wanted by the user, as then be has to re-upload it to get added back to his own timeline.

I have attached screenshots of the issue in 1.54.0. And again I'm seeing this affect both the web and mobile app.

Google shared album, using Android mobile app, I'm logged in as just a member of this shared album, not the owner, and I can select assets that I did not add to this album and I can do a 'Remove from album' function in the top toolbar:
![Screenshot_20230420-234220.png](https://user-images.githubusercontent.com/50121594/233560809-5792f38f-7023-4f90-ac2c-5def74ed6987.png)

Google shared album, using Android mobile app, continued from above, and the two selected photos that are not mine are able to be removed from the shared album (not deleted):
![Screenshot_20230420-234229.png](https://user-images.githubusercontent.com/50121594/233560748-19b3a242-2fa9-4dc6-8efb-1bfa38a0543e.png)

Note: In the above case the two photos that were removed from the album by the shared member, who is not the album owner, do not then delete from the owners timeline on the local device for that original user. It just simply removes the two selected assets from the shared album, because all members are treated equally for that function it seems in google shared albums. The function was tested in google Android mobile app and in the desktop web version.

Immich web and app 1.54.0, do not seem to allow this, unless you are the shared album owner, as when you try to use the function as a non-owner at least, to remove asset(s) from a shared album in the toolbar it doesn't display in the dialog to let you click on a button to actually be able to remove the assets: 

![Screenshot_20230420-230525.png](https://user-images.githubusercontent.com/50121594/233559629-898f2daf-4548-4311-9903-abe6738db1d1.png)

![Screenshot_20230420-230543.png](https://user-images.githubusercontent.com/50121594/233559630-489ec132-6db1-4687-a430-80f846faad68.png)

Immich 1.54.0 web logged in as a member of a shared album, not as the owner, note that there is no 'Remove from album' function in the toolbar to utilize, either in this main assets view or inside a single asset view (tested only in Chrome mobile browser for Android, in normal mobile mode and Desktop view modes):

![Screenshot_20230420-230909.png](https://user-images.githubusercontent.com/50121594/233559628-cdc0344a-862a-4513-bfe8-a958ed326f72.png)

**Comment by jrasm91** [comment]
Implemented on the web a few releases ago
